### PR TITLE
Simplices handling for Cuda backends

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR adds a simple algorithm to treat simplices on for Cuda backends.

### CODE REVIEW

File: backends/cuda/ceed-cuda-basis.c [hunk 1]
```diff
@@ -415,6 +500,67 @@ int CeedBasisApply_Cuda(CeedBasis basis, const CeedInt nelem,
   return 0;
 }
 
+int CeedBasisApplyNonTensor_Cuda(CeedBasis basis, const CeedInt nelem,
```
Comment (jeremylt) [inline backends/cuda/ceed-cuda-basis.c]: It looks like this will need the adjustment to remove "ceed-impl.h" as well.

**Comment by jeremylt** [comment]
Am I right that this follows #247?

File: backends/cuda/ceed-cuda-basis.c (hunk 1 above)
Comment (YohannDudouit) [reply backends/cuda/ceed-cuda-basis.c]: That's right, but most of these are already in cuda-shared branch, I'll rebase once the shared-memory backend gets merged into master.

**Comment by YohannDudouit** [comment]
It doesn't completely follows #247 , but was implemented afterwards, I initially thought to put the non-tensor kernels in cuda-shared backend but then I realized that the implementation was relatively trivial and unoptimized and thus decided to put it in the cuda-ref backend. I could remove all the work from the cuda-shared backend to clean this PR!

**Comment by YohannDudouit** [comment]
Refactored in #249

**Pull request closed without merging**